.github/workflows/build.yaml

```
@@ -0,0 +1,47 @@
+name: main
+
+on:
+  push:
+  pull_request:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+      uses: ./.github/workflows/testall.yaml
+      with:
+        build-artifact-name: none
+        build-artifact-path: none
+        self-build: 1
+
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    environment: PyPi-deploy
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.10.2
+      - name: Install requirements and build wheel
+        shell: bash -l {0}
+        run: |
+          python -m pip install build twine
+          python -m build .
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
+  lint:
+    if: inputs.self-build==1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

```suggestion
      - uses: actions/checkout@v3
```

It looks like `checkout@v2` generates warnings due to deprecated node.js version issue.